### DISCUSSION TITLE
PMD rules deprecated "category/java/bestpractices.xml"

### DISCUSSION THREAD

**Discussion by dutt-sanjay** [pmd/pmd#3018] [Q&A] [answered]
Below is my pmd-rules.xml. I am importing the profile using QAPlug for Intellij. Whenever I import this file i am getting below message. 

PMD Coding Rules deactivated: 298
PMD Coding Rules deprecated: 2

Additional Information:
PMD Coding Rules deprecated 
category/java/bestpractices.xml
category/java/bestpractices.xml/JUnitTestContainsTooManyAsserts

```
<?xml version="1.0"?>
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         name="Custom ruleset"
         xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">

    <description>
        Custom rules for checking JUnit test quality.
    </description>


<!--
    <rule ref="rulesets/java/basic.xml"/>
    <rule ref="rulesets/java/unusedcode.xml"/>
    <rule ref="rulesets/java/imports.xml"/>
    <rule ref="rulesets/java/strings.xml"/>
    <rule ref="rulesets/java/codesize.xml"/>
    <rule ref="rulesets/java/braces.xml"/>
    <rule ref="rulesets/java/clone.xml"/>
    <rule ref="rulesets/java/empty.xml"/>
    <rule ref="category/java/errorprone.xml">
        <exclude name="TestClassWithoutTestCases"/>
    </rule>
-->
    <rule ref="category/java/bestpractices.xml">
        <exclude name="JUnitAssertionsShouldIncludeMessage"/>
	    <exclude name="JUnitTestContainsTooManyAsserts"/>
        <exclude name="JUnit4TestShouldUseAfterAnnotation" />
        <exclude name="JUnit4TestShouldUseBeforeAnnotation" />
        <exclude name="UseVarargs" />
        <exclude name="AccessorMethodGeneration" />
    </rule>
    <rule ref="category/java/bestpractices.xml/JUnitTestContainsTooManyAsserts">
        <properties>
            <property name="maximumAsserts" value="5" />
        </properties>
    </rule>
</ruleset>
```
It's just now when I analyze code it says **no coding rules found for above rules** as the ones which are there are deprecated. 
Can you guys please look into this and tell me If I am missing anything. Thanks!

**Comment by adangel** [comment]
If I use this ruleset on the command line, I don't see any deprecated rule warning.
I'm not familiar with the intellij plugin. Maybe this plugin does a bit more? How do you configure the plugin? It seems, it introduces some kind of "coding rules" - so there seems to be a mapping between pmd rules (the rules from the ruleset) to coding rules...

**Comment by dutt-sanjay** [comment]
I am using QAPlug. This Plugin will give you feedback in your IDE. 
Plugin :- https://plugins.jetbrains.com/plugin/4597-qaplug--findbugs
Intellij QAPlug demonstration with PMD and Checkstyle YouTube Video  https://youtu.be/syAcj-FEaAU

**Comment by jslawinski** [comment, marked as answer]
@adangel Yes, you are right - there is a mapping between PMD rules and the rules in intellij. There was an issue, but has been resolved.

@dutt-sanjay you should be able to import your rules now using QAPlug-PMD 1.4.6
